src/bin/svg2png/svg2png.cpp

```diff
@@ -25,180 +25,397 @@
-            tvg::Initializer::term(tvg::CanvasEngine::Sw);
+    void terminate()
+    {
+        //Terminate ThorVG Engine
+        tvg::Initializer::term(tvg::CanvasEngine::Sw);
 
-            free(buffer);
+        //Free buffer
+        free(buffer);
+    }
 
-        } else {
-            cout << "engine is not supported" << endl;
+private:
+    void createCanvas()
+    {
+        //Canvas Engine
+        tvg::CanvasEngine tvgEngine = tvg::CanvasEngine::Sw;
+
+        //Threads Count
+        auto threads = thread::hardware_concurrency();
+
+        //Initialize ThorVG Engine
+        if (tvg::Initializer::init(tvgEngine, threads) != tvg::Result::Success) {
+            cout << "Error: Engine is not supported" << endl;
         }
+
+        //Create a Canvas
+        canvas = tvg::SwCanvas::gen();
+    }
+
+    void createBuffer(int w, int h)
+    {
+        uint32_t size = w * h;
+        //Reuse old buffer if size is enough
+        if (buffer && buffer_size >= size) return;
+
+        //Alloc or realloc buffer
+        buffer = (uint32_t*) realloc(buffer, sizeof(uint32_t) * size);
+        buffer_size = size;
     }
 
-    int setup(int argc, char **argv, size_t *w, size_t *h)
+private:
+    unique_ptr<tvg::SwCanvas> canvas = nullptr;
+    uint32_t* buffer = nullptr;
+    uint32_t buffer_size = 0;
+};
+
+struct App
+{
+public:
+    int setup(int argc, char** argv)
     {
-        char *path{nullptr};
-
-        *w = *h = 0;
-
-        if (argc > 1) path = argv[1];
-        if (argc > 2) {
-            char tmp[20];
-            char *x = strstr(argv[2], "x");
-            if (x) {
-                snprintf(tmp, x - argv[2] + 1, "%s", argv[2]);
-                *w = atoi(tmp);
-                snprintf(tmp, sizeof(tmp), "%s", x + 1);
-                *h = atoi(tmp);
+        const char* path = nullptr;
+
+        for (int i = 0, arg = 0; i < argc; i++) {
+            const char* p = argv[i];
+            if (*p == '-') {
+                //flags
+                if (p[1] == 'd') {
+                    //disable looking in subdirectories for wildcard
+                    flags |= FLAG_DISABLE_SUBDIRS;
+                } else if (p[1] == 'c') {
+                    //use ABGR8888 colorspace
+                    flags |= FLAG_USE_ABGR8888_COLORSPACE;
+                } else if (p[1] == 's') {
+                    //do not save png file. Useful for testing memory leaks
+                    flags |= FLAG_DO_NOT_SAVE_PNG;
+                } else {
+                    cout << "Warning: Unknown flag (" << p << ")." << endl;
+                }
+            } else {
+                //arguments
+                switch (arg++)
+                {
+                    case 0:
+                    {
+                        break;
+                    }
+                    case 1:
+                    {
+                        path = p;
+                        break;
+                    }
+                    case 2:
```

Comment:
It's interesting. It happened on my machine as well, but only for a single * and only without quotation marks. For *.svg and "*" or "*.svg" it doesn't.
Please find my new idea in post below. It would allow for multiple files and be would resistant to such a behavior.